modules/Sources/Features/Home/NighthawkSettings/Fiat/FiatView.swift

```diff
@@ -6,19 +6,90 @@
 //
 
 import ComposableArchitecture
+import Generated
+import Models
 import SwiftUI
+import UIComponents
 
 public struct FiatView: View {
     let store: StoreOf<Fiat>
+    let tokenName: String
     
     public var body: some View {
         WithViewStore(store, observe: { $0 }) { viewStore in
-            Text("Fiat view")
+            ScrollView([.vertical], showsIndicators: false) {
+                VStack(alignment: .leading, spacing: 16) {
+                    Text(L10n.Nighthawk.SettingsTab.FiatCurrency.title)
+                        .subtitleMedium(color: Asset.Colors.Nighthawk.parmaviolet.color)
+                    
+                    Text(L10n.Nighthawk.SettingsTab.FiatCurrency.description(tokenName))
+                        .paragraphMedium(color: .white)
+                        .multilineTextAlignment(.leading)
+                        .lineSpacing(6)
+                    
+                    RadioSelectionList(
+                        options: NighthawkSetting.FiatCurrency.allCases,
+                        selection: viewStore.$selectedFiatCurrency.animation(.none)
+                    ) { option in
+                        HStack {
+                            switch option {
+                            case .usd:
```

Comment:
[non-blocking] I would move this out of the view to a helper function, to resolve the Localization key and leave the view more readable